core/sql/optimizer/SimpleScanOptimizer.cpp

```diff
@@ -218,25 +218,14 @@ SimpleFileScanOptimizer::isLogicalSubPartitioned() const
 SearchKey *
 SimpleFileScanOptimizer::constructSearchKey()
 {
+  // We do not need to incude the flatten version of RANGE SPEC predicates 
+  // to the exePreds because SearchKey is capable of handling such predicates.
```

Comment:
My intention on the comment is that SearchKey is capable of processing
RANGE SPEC, and therefore there is no need to feed the flatten version of
it.

That complex predicates and RANGE SPEC may be partially represented in the
SeachKey probably is a known fact.  The fix tries to address it in one
aspect.

There maybe other areas that require some work, such as
cardinality/costing.

On Thu, Aug 27, 2015 at 11:08 AM, Hans Zeller <notifications@github.com>
wrote:

> In core/sql/optimizer/SimpleScanOptimizer.cpp
> <https://github.com/apache/incubator-trafodion/pull/66#discussion_r38113574>
> :
>
> > @@ -218,25 +218,14 @@ SimpleFileScanOptimizer::isLogicalSubPartitioned() const
> >  SearchKey *
> >  SimpleFileScanOptimizer::constructSearchKey()
> >  {
> > +  // We do not need to incude the flatten version of RANGE SPEC predicates
> > +  // to the exePreds because SearchKey is capable of handling such predicates.
>
> Good question, and it makes me think of two things. One is that the
> example where MDAM can't retrieve the exact set is probably one that
> involves complex predicates on two or more columns, while RangeSpec itself
> is probably simple enough. But RangeSpec AND might fall into this category.
> The second question is whether we really handle RangeSpec predicates
> correctly in MDAM code generation.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/apache/incubator-trafodion/pull/66/files#r38113574>.
>



-- 
Regards, --Qifan